src/util/yp_snprintf.c

```diff
@@ -0,0 +1,12 @@
+#include "yarp/defines.h"
+
+// In case snprintf isn't present on the system, we provide our own that simply
+// forwards to the less-safe sprintf.
+int
+yp_snprintf(char *dest, YP_ATTRIBUTE_UNUSED size_t size, const char *format, ...) {
```

Comment:
FWIW it seems snprintf might be defined on Windows, at least UCRT: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/snprintf-snprintf-snprintf-l-snwprintf-snwprintf-l?view=msvc-170
But maybe not on older Windows versions?
From https://stackoverflow.com/questions/9052224/error4error-c3861-snprintf-identifier-not-found#comment11358163_9052232 snprintf is standard in C99, so Windows should have it since CRuby requires C99 nowadays (AFAIK)